### PR TITLE
Improve type checking: wrapper types, union callables, class implements

### PR DESCRIPTION
**Pull request by mizchi**
## Summary

This PR significantly improves TypeScript type checking by adding support for several missing diagnostics and refining object-literal type handling. The changes focus on false-positive-free implementations that stay silent on cases TypeScript can't fully model.

### Type System Improvements
- **Boxed wrapper type checking**: Added detection that wrapper objects (`String`, `Number`, `Boolean`, `BigInt`, `Symbol`) are not assignable to their primitive counterparts (TS2322), preventing `var s: string = new String("")` from being incorrectly accepted.

- **Union callable support**: Implemented argument checking for union types with shared call/construct signatures. A union callee like `{ (a:number):X } | { (a:number):Y }` is now callable when all members have identical parameters, with disagreeing parameters staying silent to avoid false positives.

- **Class `implements` clause**: Parser now retains and validates `implements` declarations. The checker verifies that non-generic classes satisfy their implemented interfaces, flagging private/protected implementations and incompatible member types (TS2420).

- **Variable redeclaration checking**: Added TS2403 detection for `var` redeclarations with non-identical types. Uses order-independent union normalization to allow `var x: A | B; var x: B | A;` while catching real mismatches like `var x: C; var x: C | D;`.

### Object Literal Handling
- **Deep literal widening**: Introduced `widen_literal_deep` to widen literal types inside object/struct/array/tuple fields at binding sites, preventing false positives from over-narrow literal comparisons while preserving real category clashes.

- **Improved object type display**: Changed `Object` types from generic "type" label to rendered field shapes (e.g., `{ foo: string; bar: number }`), making mismatch messages actionable while staying false-positive-free via the new widening.

- **Structural one-way assignability**: Added `structural_oneway_assignable` that descends union targets, allowing sources like `{ p1, p2 }` to flow into `number | I2` via the interface arm.

### Diagnostic Refinements
- **Object-literal source suppression**: Added guards to suppress object-literal mismatches against non-bare-scalar targets (generics, unions, call signatures) where contextual typing isn't fully modeled, while preserving real category clashes against primitives.

- **Type-argument arity checking**: Extended to cover classes, interfaces, and enums (TS2314/TS2315), restricted to locally-declared names to avoid false positives on lib types like `Array<T>`.

- **Member override incompatibility**: Implemented `member_override_incompatible` for class method/property override validation, deciding only shapes that can be compared soundly (scalars, plain named shapes) and staying silent on generics/`any`/`unknown`.

### Parser Enhancements
- **Class member duplicate detection**: Added TS2300 detection for incompatible member declarations (data field + accessor, multiple getters/setters, getter/setter + method), while allowing legal overload sets.

- **Constructor body retention**: Parser now preserves constructor bodies so the checker can walk statements with `this` bound to the instance type.

## Implementation Details

All new checks are designed to be false-positive-free:
- Wrapper type checking only matches exact wrapper/primitive pairs
- Union callable checking requires all members to have identical parameters
- Variable redeclaration only compares when all members are identity-comparable atoms
- Object-literal suppression gates on specific imprecision classes (typeof, call signatures, non-bare-scalar targets)
- Member override checking only decides shapes it can fully model

The changes include comprehensive unit tests covering edge cases like generic interfaces, mixed scalar/object index types, and union member disagreements.

## Tooling

Added `src/cmd/tsacc` — a standalone corpus-accuracy CLI that measures TypeScript conformance recall/false-positive rates without the overhead of the full parser package test build.

https://claude.ai/code/session_0197Ev4JKxuFuk4wfw87CQUS

### CODE REVIEW

File: src/parser/parser_function.mbt [hunk 1]
```diff
@@ -2425,6 +2425,7 @@ fn Parser::parse_declare_class(
     name,
     type_params: class_type_params,
     base_names,
+    implements_names: [],
```
[the remark this answers is not in the document]
Comment (mizchi) [reply src/parser/parser_function.mbt]: Fixed in 2f5f0ea. `parse_declare_class_base_names` now returns both the `extends` base names and the `implements` names (tracked via an `in_implements` section in the same depth-aware scan), and the ambient class decl carries them, so `check_class_implements` covers `declare class` uniformly. Verified: `declare class C implements I { private x: number }` now flags TS2420; a correct ambient implementation and `declare class D extends Base implements I {}` stay silent.

---
_Generated by [Claude Code](https://claude.ai/code)_
